internal/provider/data_source_dirhash_sha256.go

```diff
@@ -0,0 +1,60 @@
+package provider
+
+import (
+	"context"
+
+	"github.com/Think-iT-Labs/dirhash/lib"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/diag"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+)
+
+func dataSourceDirhashSha256() *schema.Resource {
+	return &schema.Resource{
+		Description: "Calculate SHA256 checksum of a given directory",
+
+		ReadContext: dataSourceDirhashSha256Read,
+
+		Schema: map[string]*schema.Schema{
+			"directory": {
+				Type:        schema.TypeString,
+				Description: "Directory to hash.",
+				Required:    true,
```

Comment:
would it be better to keep the directory optional with a default value of the current Module Path? What do you think?
```suggestion
				Optional:    true
				Default:      "." // ${Path.module} 
```

Reply:
I don't think it's a good idea. I do think people might miss-use it if we do so